### ISSUE TITLE
`publish heroku` failing due to old Python version

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#1905]
Reported on Discord: https://discord.com/channels/823971286308356157/823971286941302908/1042814317118115901

```
-----> Building on the Heroku-22 stack
-----> Determining which buildpack to use for this app
-----> Python app detected
-----> Using Python version specified in runtime.txt
 !     Requested runtime 'python-3.8.10' is not available for this stack (heroku-22).
 !     For supported versions, see: https://devcenter.heroku.com/articles/python-support
 !     Push rejected, failed to compile Python app.

 !     Push failed
 ▸    Build failed
```

**Comment by simonw** [comment]
This is a strong argument for extracting the Heroku support out to a plugin - it would allow this to be fixed with a plugin release without needing to push a full release of Datasette itself.

**Comment by simonw** [comment]
The tests don't cover this pit at the moment.

Would be easier to write tests if there was a `--generate-dir` option as seen in https://datasette.io/plugins/datasette-publish-vercel

**Comment by simonw** [comment]
Tested a deploy:

```
 % datasette publish heroku fixtures.db -n datasette-issue-1905
 ›   Warning: heroku update available from 7.63.0 to 7.66.4.
 ›   Warning: heroku update available from 7.63.0 to 7.66.4.
 ›   Warning: heroku update available from 7.63.0 to 7.66.4.
Creating datasette-issue-1905... done
 ›   Warning: heroku update available from 7.63.0 to 7.66.4.
 ▸    Couldn't detect GNU tar. Builds could fail due to decompression errors
 ▸    See https://devcenter.heroku.com/articles/platform-api-deploying-slugs#create-slug-archive
 ▸    Please install it, or specify the '--tar' option
 ▸    Falling back to node's built-in compressor

-----> Building on the Heroku-22 stack
-----> Determining which buildpack to use for this app
-----> Python app detected
-----> Using Python version specified in runtime.txt
-----> Installing python-3.11.0
-----> Installing pip 22.3.1, setuptools 63.4.3 and wheel 0.37.1
-----> Installing SQLite3
-----> Installing requirements with pip
       Collecting datasette
         Downloading datasette-0.63.1-py3-none-any.whl (231 kB)
...
-----> Running post-compile hook
-----> Discovering process types
       Procfile declares types -> web

-----> Compressing...
       Done: 28M
-----> Launching...
       Released v3
       https://datasette-issue-1905.herokuapp.com/ deployed to Heroku

Starting November 28th, 2022, free Heroku Dynos, free Heroku Postgres, and free Heroku Data for Redis® will no longer be available.

If you have apps using any of these resources, you must upgrade to paid plans by this date to ensure your apps continue to run and to retain your data. For students, we will announce a new program by the end of September. Learn more at https://blog.heroku.com/next-chapter
```
I had to then pay for the dino because I'd run out of free hours.

https://datasette-issue-1905.herokuapp.com/-/versions shows:

```json
{
    "python": {
        "version": "3.11.0",
        "full": "3.11.0 (main, Oct 24 2022, 21:34:02) [GCC 11.2.0]"
    },
    "datasette": {
        "version": "0.63.1"
    }
```

**Comment by simonw** [comment]
Used it to deploy this: https://fivethirtyeight.datasettes.com/-/versions
